R/interval_formatting.R

```diff
@@ -95,8 +95,10 @@ gen_end <- function(dt, id_cols, col_stem, right_most_endpoint = Inf) {
   assertthat::assert_that(!end_col %in% names(dt),
                           msg = paste0("'", end_col,
                                        "' column already in `dt`"))
-  assertive::assert_is_numeric(dt[[start_col]])
-  assertive::assert_all_are_not_na(dt[[start_col]])
+  assertthat::assert_that(
```

Comment:
From `profvis` I was able to identify that `assertive:: assert_all_are_not_na` was unusually slow when we expect this to be simple and quick.

I used the microbenchmark package to measure how fast each of these two assertion methods are and double check that this change will be substantially faster. http://adv-r.had.co.nz/Performance.html#microbenchmarking

```
x <- rep(0, 10000000)

microbenchmark::microbenchmark(
  assertive::assert_all_are_not_na(x),
  assertthat::assert_that(
    assertive::is_numeric(x),
    all(!is.na(x))
  ),
  times = 10, unit = "s"
)
Unit: seconds
                                                              expr        min         lq       mean     median         uq        max neval
                               assertive::assert_all_are_not_na(x) 3.63133398 3.73356033 3.81967985 3.83751208 3.85160660 4.16463184    10
 assertthat::assert_that(assertive::is_numeric(x), all(!is.na(x))) 0.03361764 0.03924905 0.05283393 0.05686381 0.06528024 0.06621265    10

```

Reply:
Actually I guess `bench` is a better maintained package for doing this.